test/SILOptimizer/specialize_unconditional_checked_cast.swift

```diff
@@ -97,9 +97,13 @@ ArchetypeToConcreteConvertUInt8(t: c)
 ArchetypeToConcreteConvertUInt8(t: f)
 
 // x -> x where x is not a class.
+//
+// TODO: Why is the optimizer cloning twice here.
+//
 // CHECK-LABEL: sil shared [noinline] @$s37specialize_unconditional_checked_cast31ArchetypeToConcreteConvertUInt8{{[_0-9a-zA-Z]*}}3Not{{.*}}Tg5 : $@convention(thin) (NotUInt8) -> NotUInt8 {
 // CHECK: bb0
 // CHECK-NEXT: debug_value %0
+// CHECK-NEXT: debug_value %0
```

Comment:
@adrian-prantl this was the test failure. It happens during optimization. I don't think it is interesting b/c it is in -O code.

Reply:
Adrian and I tracked this down to a bug in mem2reg. I restore this test in the last commit of this PR and in that commit return the test back to its former form.